src/promptflow/promptflow/_trace/_start_trace.py

```diff
@@ -60,8 +60,8 @@ def start_trace(*, session: typing.Optional[str] = None, **kwargs):
     env_attributes = json.loads(env_trace_context).get("attributes") if env_trace_context else {}
     experiment = env_attributes.get(ContextAttributeKey.EXPERIMENT, None)
     ref_line_run_id = env_attributes.get(ContextAttributeKey.REFERENCED_LINE_RUN_ID, None)
-    if ref_line_run_id is not None:
-        operation_context._add_otel_attributes(SpanAttributeFieldName.REFERENCED_LINE_RUN_ID, ref_line_run_id)
+    # Set reference line run id even if it's None to avoid stale value set by previous node
+    operation_context._add_otel_attributes(SpanAttributeFieldName.REFERENCED_LINE_RUN_ID, ref_line_run_id)
```

Comment:
will executor ignore those `None` fields?

Reply:
![image](https://github.com/microsoft/promptflow/assets/38847871/2b0c3007-bd41-46a4-b3df-d6a193c1db11)
 
not sure if this will add null fields here, we might need @thy09 comment

Reply:
just offline synced with Heyi, he is not sure the behavior inside OTel, to promise this, we better touch `_get_otel_attributes` to filter those fields with null value.

Reply:
updated by remove none